Yoast/wordpress-seo
# Remove htmlspecialchars

## Context
Described in issue https://github.com/Yoast/wordpress-seo/issues/18953 

`htmlspecialchars` breaks urls for images that have GET parameters.

## Summary


This PR can be summarized in the following changelog entry:

* Allow urls with query parameters in some upload fields

## Relevant technical choices:

* `htmlspecialchars` modifies the `$submitted_url` and potentially breaks good urls. With that in mind, the line right after already sanitizes urls very thoroughly. There is no need for this.

## Test instructions
First add this snippet to functions.php (or somewhere that will run for sure):

```
/**
 * Pseudo set image quality. It just adds get params to image urls. In real life this would be far more complex.
 */
add_filter(
	'wp_get_attachment_url',
	function ( $attachment_url ) {
		$image_url = add_query_arg(
			[
				'quality' => 90,
				'grain'   => 0.5,
			],
			$attachment_url
		);
		return $image_url;
	}
);
```

With that in place, go to to the Yoast SEO menu -> General -> First-time configuration -> proceed to Site Representation -> add any image to the organization logo -> Save and continue should fail to work and prevent you from going to the next step (Social Profiles).


### Test instructions for QA when the code is in the RC
See above


## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

* Images saved to different options, specifically: company_logo, person_logo, open_graph_frontpage_image, social-image-url-

## Documentation


## Quality assurance


Fixes #18953
